Added documentation that there will always be at least one target, even with `main: false` (#761)
parcel-bundler/website: src/configuration/package-json.md

If the `browser` field is an [object](/features/module-resolution/#package.json-browser-field), `package.json#browser[pkgName]` can be used instead of `package.json#browser`.

> Note that there will always be at least one target, even with `main: false`(This doesn't mean that Parcel will do nothing when running parcel build src/index.js but that the default target will be used instead of main.)

Reviewer: ```suggestion
Note that there will always be at least one target, even with `main: false`. (So the default target will be used instead of `main`.)
```

Author: @mischnic I have implemented the suggested changes. please review the pull request.
